velox/common/dynamic_registry/README.md

```diff
@@ -0,0 +1,22 @@
+# Velox: Dynamically Loading Registry Libraries in C++
+
+This library adds the ability to load User Defined Functions (UDFs), connectors, or types without having to fork and build Prestissimo, through the use of shared libraries that a Prestissimo worker can access. These are to be loaded on launch of the Presto server. The Presto server searches for any .so or .dylib files and loads them using this library.
+
+## Getting started
+1. Create a cpp file for your dynamic library
+For dynamically loaded function registration, the format followed is mirrored of that of built-in function registration with some noted differences. Using [MyDynamicTestFunction.cpp](tests/MyDynamicTestFunction.cpp) as an example, the function uses the extern "C" keyword to protect against name mangling. A registry() function call is also necessary here.
+
+2. Register functions dynamically by creating .dylib or .so shared libraries and dropping them in a plugin directory
+These shared libraries may be made using CMakeLists like the following:
+
+```
+add_library(name_of_dynamic_fn SHARED TestFunction.cpp)
+target_link_libraries(name_of_dynamic_fn PRIVATE xsimd fmt::fmt velox_expression)
+```
+
+3. In the Prestissimo worker's config.properties file, set the plugin.dir property
```

Comment:
This is not relevant in Velox. And also since its not used anywhere in the current code, its hard to put this in picture.